sbom/formatted_reader.go

```diff
@@ -78,6 +87,61 @@ func (f *FormattedReader) Read(b []byte) (int, error) {
 			}
 		}
 
+		// Makes SPDX SBOM more reproducible, see
+		// https://github.com/paketo-buildpacks/packit/issues/368 for more details.
+		if f.format.ID() == "spdx-2-json" {
+			hash, err := hashstructure.Hash(f.sbom.syft, hashstructure.FormatV2, nil)
+			if err != nil {
+				// not tested
+				return 0, fmt.Errorf("failed to modify SPDX SBOM for reproducibility: %w", err)
+			}
+			hashBytes := make([]byte, binary.MaxVarintLen64)
+			binary.PutUvarint(hashBytes, hash)
```

Comment:
If you wanted remove the need to pull in this extra third-party library, you could just use stdlib. Maybe something like the following:

```go
var spdxOutput map[string]interface{}
err = json.Unmarshal(output, &spdxOutput)
if err != nil {
	return 0, fmt.Errorf("failed to modify SPDX SBOM for reproducibility: %w", err)
}

delete(spdxOutput, "documentNamespace")
delete(spdxOutput, "creationInfo")

hash := sha1.New()
err = json.NewEncoder(hash).Encode(spdxOutput)
if err != nil {
	return 0, fmt.Errorf("failed to checksum SPDX document: %w", err)
}

checksumSha1 := hash.Sum(nil)
```

Reply:
I personally am not too concerned about pulling in `mitchellh/hashstructure`.

Reply:
I ended up going with the other approach in my spike because it felt weird to create a SHA that doesn't really correspond to anything -- it's not the SHA of the final SBOM, nor is it the SHA of the original SBOM or the package list in the SBOM. But I do like that this approach avoids pulling in a third-party lib.
 @ForestEckhardt do you have thoughts on this?

Reply:
FWIW `mitchellh/hashstructure` doesn't have external dependencies, we're not adding much more bloat to `packit`.

Reply:
I have just pushed an implementation that does not use  `mitchellh/hashstructure` that I think might be more straight forward. Please take a look and let me known what you think!